modules/prebidServerBidAdapter/ortbConverter.js

```diff
@@ -57,7 +57,7 @@ const PBS_CONVERTER = ortbConverter({
       let {s2sBidRequest, requestedBidders, eidPermissions} = context;
       const request = buildRequest(imps, proxyBidderRequest, context);
 
-      request.tmax = s2sBidRequest.s2sConfig.timeout;
+      request.tmax = s2sBidRequest.s2sConfig.ortb2?.ext?.tmaxmax || s2sBidRequest.s2sConfig.timeout;
```

Comment:
my reading of the [issue](https://github.com/prebid/Prebid.js/issues/10964):

> This is great, but we have a module that would like to know the actual client-side timeout -- the timeout passed to pbjs.requestBids.

> The ortbConverter library should check for the existence of ext.tmaxmax and if not there, set it to the timeout value passed to requestBids()

is that this should be a new field, `request.ext.tmaxmax`, which by default should be the client timeout, optionally overridable with `ortb2.ext.tmaxmax`

which I think in code translates to

```suggestion
      request.ext.tmaxmax = request.ext.tmaxmax ?? proxyBidderRequest.timeout
```

Reply:
Hmmm ... yes, I think I agree with you @dgirardi.  

Belated question: do we think that `tmaxmax` is something that a SSP endpoint (the place a bid request will get POST-ed to) needs to worry about?  Or is this just to help the server-side make a smart decision about `tmax`?  I'm assuming since `tmaxmax` is not part of oRTB the answer is "no", but thought I'd ask.

Reply:
@dgirardi I'm having trouble writing a test to verify that this works.  

Where would `ortbConverter.js` look to find out if a publisher had made this call:
```
pbjs.setConfig({
   ortb2: {
      ext: {
         tmaxmax: 888
      }
   }
});
```

Would it be in `proxyBidderRequest`?  That object is undefined right now in the test I'm running, but if that is where a publisher-specified `tmaxmax` value would go I could mock one up.

Reply:
roughly, `ortb2` comes in as `s2sBidRequest.ortb2Fragments.global` here:

https://github.com/prebid/Prebid.js/blob/e07451212d8b708fe662031f8a4a1e159b469b49/modules/prebidServerBidAdapter/index.js#L454

passed down here:

https://github.com/prebid/Prebid.js/blob/e07451212d8b708fe662031f8a4a1e159b469b49/modules/prebidServerBidAdapter/ortbConverter.js#L262

and finally here:

https://github.com/prebid/Prebid.js/blob/e07451212d8b708fe662031f8a4a1e159b469b49/modules/prebidServerBidAdapter/ortbConverter.js#L172

which is how, in the code at the top of this thread, it ends up filling `request.ext.tmaxmax` after you run `buildRequest`.

"roughly" because there's more than one way to configure `ortb2` - and the PBS adapter is peculiar in that it also has to deal with bidder-specific configuration separately from global configuration (hence the `.global`)

in practice for testing that means manipulating `ortb2Fragments.global` in the first argument to `callBids` - here's an example:

https://github.com/prebid/Prebid.js/blob/e07451212d8b708fe662031f8a4a1e159b469b49/test/spec/modules/prebidServerBidAdapter_spec.js#L2614-L2621

Reply:
This is great - thank you so much!  I'll try some of this out later this afternoon and let you know.

Reply:
Addressed with commit `b3b3ca4`